tests/install_upgrade_operators/network_policy/test_network_policy_components.py

```diff
@@ -0,0 +1,307 @@
+            )
+    yield {
+        "deployed_client_pod": deployed_client_pod,
+        "service_ip": service_ip,
+        "service_port": service_port,
+        "hco_namespace": hco_namespace,
+    }
+
+
+@pytest.fixture
+def egress_connectivity_baseline(
+    pods_by_network_policy, hco_namespace, admin_client, network_policy_match_labels, deployed_server_service
+):
+    """
+    Test preparation for egress tests: Apply allow-all NetworkPolicy, verify traffic works, then remove it.
+    This ensures that without restrictive policies, the connection should succeed.
+    """
+    with AllowAllNetworkPolicy(
+        name="test-allow-all-traffic",
+        namespace=hco_namespace.name,
+        client=admin_client,
+        match_labels=network_policy_match_labels,
+    ):
+        pod_name = pods_by_network_policy[0]
+        service_ip = (
+            deployed_server_service.instance.spec.get("clusterIPs")
+            or [deployed_server_service.instance.spec.get("clusterIP")]
+        )[0]
+        if ":" in service_ip:
+            pytest.skip(f"This test does not support sending traffic to IPv6: {service_ip}")
+
+        # Get Pod object to use execute() method
+        test_pod = Pod(client=admin_client, namespace=hco_namespace.name, name=pod_name)
+        try:
+            test_pod.execute(
+                command=shlex.split(f"curl -sS --connect-timeout 5 http://{service_ip}:{TEST_SERVER_PORT}")
+            )
+        except Exception as e:
+            pytest.fail(
+                f"Test preparation FAILED (egress): With allow-all NetworkPolicy, "
+                f"connection should succeed but got error: {e}"
+            )
+
+    yield {
+        "pods_by_network_policy": pods_by_network_policy,
+        "hco_namespace": hco_namespace,
+        "server_service_ip": service_ip,
+    }
+
+
+@pytest.mark.polarion("CNV-12305")
+@pytest.mark.parametrize("network_policy_name", [KUBEVIRT_CONSOLE_PLUGIN_NP])
+def test_console_plugin_network_policy_blocks_ingress_unauthorized_traffic(
+    network_policy_name, ingress_connectivity_baseline
+):
+    """
+    Test that KUBEVIRT_CONSOLE_PLUGIN_NP blocks unauthorized traffic.
+
+    It uses a test pod with curl to try to reach the service IP+PORT,
+    which should be blocked by the NetworkPolicy.
+    """
+    deployed_client_pod = ingress_connectivity_baseline["deployed_client_pod"]
+    service_ip = ingress_connectivity_baseline["service_ip"]
+    service_port = ingress_connectivity_baseline["service_port"]
+
+    try:
+        deployed_client_pod.execute(
+            command=shlex.split(f"curl -sS -k --connect-timeout 5 https://{service_ip}:{service_port}")
+        )
+        pytest.fail(
+            f"NetworkPolicy test FAILED: {network_policy_name} connection should be blocked "
+            f"by NetworkPolicy but succeeded"
+        )
+    except Exception:
+        # Expected: connection should be blocked by NetworkPolicy
+        pass
```

Comment:
you should not succeed on all possible exceptions.. for negative tests see examples with `pytest.raises`, i.e.:

https://github.com/RedHatQE/openshift-virtualization-tests/blob/main/tests/virt/node/general/test_machinetype.py#L190